babel/messages/frontend.py

```diff
@@ -1111,34 +1111,63 @@ def parse_mapping(fileobj, filename=None):
 
     return method_map, options_map
 
+def _parse_spec(s):
```

Comment:
```suggestion
def _parse_spec(s: str) -> tuple[int | None, tuple[int, ...]]:
    """Parse a keyword spec string.
    
    See the documentation for `parse_keywords`
    for documentation on the format.
    """
```
or similar might be nice here?